views/default/page/default.php

```diff
@@ -1,58 +1,43 @@
 <?php
+
 /**
- * Elgg pageshell
- * The standard HTML page shell that everything else fits into
- *
- * @package Elgg
- * @subpackage Core
+ * Renders a standard HTML page shell
  *
+ * @uses $vars['html_attrs']  Attributes of the <html> tag
  * @uses $vars['head']        Parameters for the <head> element
  * @uses $vars['body_attrs']  Attributes of the <body> tag
+ * @uses $vars['page_attrs']  Attributes of the .elgg-page container
+ * @uses $vars['title']       Title of the page
  * @uses $vars['body']        The main content of the page
  * @uses $vars['sysmessages'] A 2d array of various message registers, passed from system_messages()
  */
-
 // render content before head so that JavaScript and CSS can be loaded. See #4032
+$sections = [
+	'messages' => elgg_view('page/elements/messages', [
+		'object' => $vars['sysmessages'],
+	]),
+	'topbar' => elgg_view('page/elements/topbar', $vars),
+	'header' => elgg_view('page/elements/header', $vars),
+	'navbar' => elgg_view('page/elements/navbar', $vars),
+	'body' => elgg_view('page/elements/body', $vars),
+	'footer' => elgg_view('page/elements/footer', $vars),
+];
+
+$page = '';
+foreach ($sections as $section => $content) {
+	$page .= elgg_view("page/elements/$section/before", $vars);
```

Comment:
I suppose we can drop it. If really wanted, you can extend `page/elements/section` and sniff the section name before outputting content.